// @expo/xcpretty/build/Parser.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Parser = void 0;
const path_1 = require("path");
const Matchers_1 = require("./Matchers");
const switchRegex_1 = require("./switchRegex");
function unescaped(args) {
    return args.map(v => v.replace(/\\/g, ''));
}
class Parser {
    constructor(formatter) {
        this.formatter = formatter;
        this.formattedSummary = false;
        this.failures = {};
        this.linkerFailure = {
            files: [],
        };
        this.currentIssue = {};
    }
    parse(text) {
        const matchedErrorResults = this.updateErrorState(text);
        if (this.shouldFormatWarning()) {
            return this.formatCompileWarning();
        }
        if (this.shouldFormatError()) {
            return this.formatCompileError();
        }
        if (matchedErrorResults) {
            return '';
        }
        const matchedLinkerResults = this.updateLinkerFailureState(text);
        if (this.shouldFormatUndefinedSymbols()) {
